src/peer_manager.rs

```diff
@@ -1528,6 +1544,14 @@ impl PeerManager {
         }
     }
 
+    /// Removes the given entry from the routing_table or peer_map, returns the removal details
+    pub fn purge_out_of_sync_peer(&mut self,
+                                  name: &XorName)
+                                  -> Result<RemovalDetails<XorName>, RoutingTableError> {
+        let _ = self.peer_map.remove_by_name(name);
```

Comment:
Yes, why not do the cleanup inside `get_routing_peer_details` (which should probably get a more descriptive name anyway) and only pass the removal details back to `Node`, so that we don't have to call another peer manager method from there to actually remove that peer.